File: java/stringRepeat.java
// repeats a substring a number of times

public class stringRepeat {
  public static String repeatStr(final int numRepeat, final String subString) {
    return subString.repeat(numRepeat);
  }

  public static void main(String[] args) {
    System.out.println(repeatStr(3, "ha"));
  }
}

File: kt/StringRepeat.kt
// repeat a string r number of times

fun repeatStr(r: Int, str: String) : String = str.repeat(r)

fun main() {
    println(repeatStr(4, "a"))     // expected: aaaa
    println(repeatStr(3, "Hello")) // expected: HelloHelloHello
    println(repeatStr(5, ""))      // expected: 
    println(repeatStr(0, "kata"))  // expected: 
    println(repeatStr(0, ""))      // expected: 
    println(repeatStr(6, "I"))     // expected: IIIIII
    println(repeatStr(5, "Hello")) // expected: HelloHelloHelloHelloHello
}
